doc: Combined TSC

Fishrock123 opened the pull request:
Updated, the old version can be found at https://github.com/nodejs/io.js/pull/1710.

Everyone gave their consent in that thread, except for @mhdawson, who I've removed until I get a "+1" from.

R= @mikeal / @nodejs/tsc 

Note: the list is currently in no particular order at all.

mikeal commented:
+1

bnoordhuis commented:
Commit log could be a little better though.  Combined TSC _what_?

rvagg commented:
I'd like to see @mhdawson back in there, he's in the "official" list https://nodejs.org/about/organization/ and attends all the meetings

mikeal commented:
> I'd like to see @mhdawson back in there, he's in the "official" list https://nodejs.org/about/organization/ and attends all the meetings

Agreed, that was probably an oversight from that day or so we were worried that we might be over the limit for IBM or something.

Fishrock123 commented:
Updated PTAL. It's changed quite a bit so I re-committed it under myself instead.

Ordered roughly in order of addition (since the collaborators still are). And added @mhdawson back, even though we've never actually gotten consent from him on that.

@bnoordhuis does this commit message look good?

Fishrock123 commented:
Enough waiting on this, landed in 4e92dbc26bdf40a1057be0e2e2b85742e1854c68
